internal/debug/flags.go

```diff
@@ -151,12 +152,23 @@ func Setup(ctx *cli.Context) error {
 	useColor := logFile == "" && os.Getenv("TERM") != "dumb" && (isatty.IsTerminal(os.Stderr.Fd()) || isatty.IsCygwinTerminal(os.Stderr.Fd()))
 
 	var logfmt log.Format
-	if ctx.Bool(logjsonFlag.Name) {
+	switch ctx.String(logFormatFlag.Name) {
+	case "json":
 		logfmt = log.JSONFormat()
-	} else if ctx.Bool(logfmtFlag.Name) {
+	case "logfmt":
 		logfmt = log.LogfmtFormat()
-	} else {
+	case "terminal":
 		logfmt = log.TerminalFormat(useColor)
+	case "":
+		// Retain backwards compatibility with `--log.json` flag if `--log.format` not set
+		if ctx.Bool(logjsonFlag.Name) {
+			logfmt = log.JSONFormat()
```

Comment:
We usually print a deprecation-warning when someone is using a flag we wish to no longer support, something like this: 
```suggestion
		if ctx.Bool(logjsonFlag.Name) {
			log.Warning("The flag '--log.json' is deprecated, please use '--log.format=json' instead")
			logfmt = log.JSONFormat()
```

Reply:
Great idea. Added.